source/agora/node/Ledger.d

```diff
@@ -281,7 +281,21 @@ public class Ledger
         const ulong next_height = this.getBlockHeight() + 1;
         auto utxo_finder = this.utxo_set.getUTXOFinder();
 
-        this.enroll_man.pool.getEnrollments(data.enrolls);
+        // FIXME: This is a pretty simple way to select enrollments
+        // What about nodes that can't be re-enrolled yet ?
+        // But it fixes some of the failures we saw in the CI.
+        // TODO: Put a check higher-up that a block generated from the
+        // nomination data can be added.
+        Enrollment[] candidates;
+        this.enroll_man.pool.getEnrollments(candidates);
+        foreach (ref c; candidates)
+        {
+            if (this.enroll_man.getEnrolledHeight(c.utxo_key) == ulong.max)
+                data.enrolls ~= c;
+            else
+                this.enroll_man.pool.remove(c.utxo_key);
```

Comment:
Right, now I see what you meant by removing from the pool.

I guess we also need some mechanism of putting the enrollments back into the pool if the candidate set here is not the same set that becomes externalized.

Reply:
Should we file an issue for this?

Reply:
#900

Reply:
Note: With a periodic cleanup, we could also simply remove this branch

Reply:
I don't think this was correct. The items should be removed from the pool after a block is externalized, not during the nomination process..

My Quorum PR failing. I'm still investigating why though.

Reply:
Well the idea with the pool is that we shouldn't *need* the items in the pool. Evicting items from the pool too early simply means we'll have less validators to propose, and as a results, the blocks we nominate are less likely to be chosen (once the consensus rules are implemented).

Reply:
I can see why it's an issue for my PR. It's because I'm reusing the same UTXO key. But this should be supported as mentioned in #824.

Also I don't quite understand your comment.

Reply:
I don't understand the reason why the candidates should be removed from the pool before the consensus is made. What I'm concerned about is the case a consensus fails with enrollments removed.